--- a/code/BoostWorkaround/boost/tuple/tuple.hpp	2011-05-31 20:14:18.000000000 +0930
+++ b/code/BoostWorkaround/boost/tuple/tuple.hpp	2014-09-10 08:25:06.000000000 +0930
@@ -175,13 +175,13 @@
 		// Get a specific tuple element
 		template <unsigned N>
 		typename detail::type_getter<T0,0,typename very_long::next_type, N>::type& get ()	{
-			return m.get<N>();
+			return m.template get<N>();
 		}
 
 		// ... and the const version
 		template <unsigned N>
 		const typename detail::type_getter<T0,0,typename very_long::next_type, N>::type& get () const	{
-			return m.get<N>();
+			return m.template get<N>();
 		}
 
 
@@ -208,14 +208,14 @@
 	template <unsigned N,typename T0,typename T1,typename T2,typename T3,typename T4>
 	inline typename tuple<T0,T1,T2,T3,T4>::very_long::template type_getter<N>::type& get (
 			tuple<T0,T1,T2,T3,T4>& m)	{
-			return m.get<N>();
+			return m.template get<N>();
 		}
 
 	// ... and the const version
 	template <unsigned N,typename T0,typename T1,typename T2,typename T3,typename T4>
 	inline const typename tuple<T0,T1,T2,T3,T4>::very_long::template type_getter<N>::type& get (
 			const tuple<T0,T1,T2,T3,T4>& m)	{
-			return m.get<N>();
+			return m.template get<N>();
 		}
 
 	// Constructs a tuple with 5 elements
@@ -224,11 +224,11 @@
 		const T1& t1,const T2& t2,const T3& t3,const T4& t4) {
 
 		tuple <T0,T1,T2,T3,T4> t;
-		t.get<0>() = t0;
-		t.get<1>() = t1;
-		t.get<2>() = t2;
-		t.get<3>() = t3;
-		t.get<4>() = t4;
+		t.template get<0>() = t0;
+		t.template get<1>() = t1;
+		t.template get<2>() = t2;
+		t.template get<3>() = t3;
+		t.template get<4>() = t4;
 		return t;
 	}
 
@@ -237,10 +237,10 @@
 	inline tuple <T0,T1,T2,T3> make_tuple (const T0& t0,
 		const T1& t1,const T2& t2,const T3& t3) {
 		tuple <T0,T1,T2,T3> t;
-		t.get<0>() = t0;
-		t.get<1>() = t1;
-		t.get<2>() = t2;
-		t.get<3>() = t3;
+		t.template get<0>() = t0;
+		t.template get<1>() = t1;
+		t.template get<2>() = t2;
+		t.template get<3>() = t3;
 		return t;
 	}
 
@@ -249,9 +249,9 @@
 	inline tuple <T0,T1,T2> make_tuple (const T0& t0,
 		const T1& t1,const T2& t2) {
 		tuple <T0,T1,T2> t;
-		t.get<0>() = t0;
-		t.get<1>() = t1;
-		t.get<2>() = t2;
+		t.template get<0>() = t0;
+		t.template get<1>() = t1;
+		t.template get<2>() = t2;
 		return t;
 	}
 
@@ -260,8 +260,8 @@
 	inline tuple <T0,T1> make_tuple (const T0& t0,
 		const T1& t1) {
 		tuple <T0,T1> t;
-		t.get<0>() = t0;
-		t.get<1>() = t1;
+		t.template get<0>() = t0;
+		t.template get<1>() = t1;
 		return t;
 	}
 
@@ -269,7 +269,7 @@
 	template <typename T0>
 	inline tuple <T0> make_tuple (const T0& t0) {
 		tuple <T0> t;
-		t.get<0>() = t0;
+		t.template get<0>() = t0;
 		return t;
 	}
 
