diff --git a/src/sdl/sdlimage.cpp b/src/sdl/sdlimage.cpp
index cab5eff..4800ea7 100644
--- a/src/sdl/sdlimage.cpp
+++ b/src/sdl/sdlimage.cpp
@@ -147,7 +147,8 @@ namespace gcn
 
         SDL_Surface *tmp;
 
-        if (hasAlpha)
+        // Removed, SDL2
+        /*if (hasAlpha)
         {
             tmp = SDL_DisplayFormatAlpha(mSurface);
             SDL_FreeSurface(mSurface);
@@ -158,7 +159,7 @@ namespace gcn
             tmp = SDL_DisplayFormat(mSurface);
             SDL_FreeSurface(mSurface);
             mSurface = NULL;
-        }
+        }*/
 
         if (tmp == NULL)
         {
@@ -170,10 +171,12 @@ namespace gcn
             SDL_SetColorKey(tmp, SDL_SRCCOLORKEY,
                             SDL_MapRGB(tmp->format,255,0,255));
         }
-        if (hasAlpha)
+        
+        // Removed, SDL2
+        /*if (hasAlpha)
         {
             SDL_SetAlpha(tmp, SDL_SRCALPHA, 255);
-        }
+        }*/
 
         mSurface = tmp;
     }
